build fails on ppc64le (TARGET=POWER8) with GCC 14 due incompatible pointer types

Builds of release 0.3.26 fail on ppc64le (TARGET=POWER8) with GCC 14 due incompatible pointer types. This is on Fedora Rawhide. All other platforms in Fedora (aarch64, x86_64, s390x) build OK.
````
...
In file included from ../common.h:849,
                 from gemmt.c:37:
gemmt.c: In function ‘sbgemmt_’:
../common_stackalloc.h:59:10: error: assignment to ‘bfloat16 *’ {aka ‘short unsigned int *’} from incompatible pointer type ‘float *’ [-Wincompatible-pointer-types]
   59 |   BUFFER = stack_alloc_size ? stack_buffer : (TYPE *)blas_memory_alloc(1);
      |          ^
gemmt.c:481:25: note: in expansion of macro ‘STACK_ALLOC’
  481 |                         STACK_ALLOC(buffer_size, FLOAT, buffer);
      |                         ^~~~~~~~~~~
gemmt.c:504:70: error: passing argument 5 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  504 |                                 (gemv[(int)transa]) (j, k, 0, alpha, aa, lda,
      |                                                                      ^~
      |                                                                      |
      |                                                                      bfloat16 * {aka short unsigned int *}
gemmt.c:504:70: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:505:54: error: passing argument 7 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  505 |                                                      bb, incb, cc, 1, buffer);
      |                                                      ^~
      |                                                      |
      |                                                      bfloat16 * {aka short unsigned int *}
gemmt.c:505:54: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:505:71: error: passing argument 11 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  505 |                                                      bb, incb, cc, 1, buffer);
      |                                                                       ^~~~~~
      |                                                                       |
      |                                                                       bfloat16 * {aka short unsigned int *}
gemmt.c:505:71: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:507:70: error: passing argument 5 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  507 |                                 (gemv[(int)transa]) (k, j, 0, alpha, aa, lda,
      |                                                                      ^~
      |                                                                      |
      |                                                                      bfloat16 * {aka short unsigned int *}
gemmt.c:507:70: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:508:54: error: passing argument 7 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  508 |                                                      bb, incb, cc, 1, buffer);
      |                                                      ^~
      |                                                      |
      |                                                      bfloat16 * {aka short unsigned int *}
gemmt.c:508:54: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:508:71: error: passing argument 11 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  508 |                                                      bb, incb, cc, 1, buffer);
      |                                                                       ^~~~~~
      |                                                                       |
      |                                                                       bfloat16 * {aka short unsigned int *}
gemmt.c:508:71: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
../common_stackalloc.h:59:10: error: assignment to ‘bfloat16 *’ {aka ‘short unsigned int *’} from incompatible pointer type ‘float *’ [-Wincompatible-pointer-types]
   59 |   BUFFER = stack_alloc_size ? stack_buffer : (TYPE *)blas_memory_alloc(1);
      |          ^
gemmt.c:570:25: note: in expansion of macro ‘STACK_ALLOC’
  570 |                         STACK_ALLOC(buffer_size, FLOAT, buffer);
      |                         ^~~~~~~~~~~
gemmt.c:593:70: error: passing argument 5 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  593 |                                 (gemv[(int)transa]) (j, k, 0, alpha, a, lda, bb,
      |                                                                      ^
      |                                                                      |
      |                                                                      bfloat16 * {aka short unsigned int *}
gemmt.c:593:70: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:593:78: error: passing argument 7 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  593 |                                 (gemv[(int)transa]) (j, k, 0, alpha, a, lda, bb,
      |                                                                              ^~
      |                                                                              |
      |                                                                              bfloat16 * {aka short unsigned int *}
gemmt.c:593:78: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:594:67: error: passing argument 11 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  594 |                                                      incb, cc, 1, buffer);
      |                                                                   ^~~~~~
      |                                                                   |
      |                                                                   bfloat16 * {aka short unsigned int *}
gemmt.c:594:67: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:596:70: error: passing argument 5 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  596 |                                 (gemv[(int)transa]) (k, j, 0, alpha, a, lda, bb,
      |                                                                      ^
      |                                                                      |
      |                                                                      bfloat16 * {aka short unsigned int *}
gemmt.c:596:70: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:596:78: error: passing argument 7 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  596 |                                 (gemv[(int)transa]) (k, j, 0, alpha, a, lda, bb,
      |                                                                              ^~
      |                                                                              |
      |                                                                              bfloat16 * {aka short unsigned int *}
gemmt.c:596:78: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:597:67: error: passing argument 11 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  597 |                                                      incb, cc, 1, buffer);
      |                                                                   ^~~~~~
      |                                                                   |
      |                                                                   bfloat16 * {aka short unsigned int *}
gemmt.c:597:67: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
...
In file included from ../common.h:849,
                 from gemmt.c:37:
gemmt.c: In function ‘cblas_sbgemmt’:
../common_stackalloc.h:59:10: error: assignment to ‘bfloat16 *’ {aka ‘short unsigned int *’} from incompatible pointer type ‘float *’ [-Wincompatible-pointer-types]
   59 |   BUFFER = stack_alloc_size ? stack_buffer : (TYPE *)blas_memory_alloc(1);
      |          ^
gemmt.c:481:25: note: in expansion of macro ‘STACK_ALLOC’
  481 |                         STACK_ALLOC(buffer_size, FLOAT, buffer);
      |                         ^~~~~~~~~~~
gemmt.c:505:71: error: passing argument 11 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  505 |                                                      bb, incb, cc, 1, buffer);
      |                                                                       ^~~~~~
      |                                                                       |
      |                                                                       bfloat16 * {aka short unsigned int *}
gemmt.c:505:71: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:508:71: error: passing argument 11 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  508 |                                                      bb, incb, cc, 1, buffer);
      |                                                                       ^~~~~~
      |                                                                       |
      |                                                                       bfloat16 * {aka short unsigned int *}
gemmt.c:508:71: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
../common_stackalloc.h:59:10: error: assignment to ‘bfloat16 *’ {aka ‘short unsigned int *’} from incompatible pointer type ‘float *’ [-Wincompatible-pointer-types]
   59 |   BUFFER = stack_alloc_size ? stack_buffer : (TYPE *)blas_memory_alloc(1);
      |          ^
gemmt.c:570:25: note: in expansion of macro ‘STACK_ALLOC’
  570 |                         STACK_ALLOC(buffer_size, FLOAT, buffer);
      |                         ^~~~~~~~~~~
gemmt.c:594:67: error: passing argument 11 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  594 |                                                      incb, cc, 1, buffer);
      |                                                                   ^~~~~~
      |                                                                   |
      |                                                                   bfloat16 * {aka short unsigned int *}
gemmt.c:594:67: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
gemmt.c:597:67: error: passing argument 11 of ‘gemv[transa]’ from incompatible pointer type [-Wincompatible-pointer-types]
  597 |                                                      incb, cc, 1, buffer);
      |                                                                   ^~~~~~
      |                                                                   |
      |                                                                   bfloat16 * {aka short unsigned int *}
gemmt.c:597:67: note: expected ‘float *’ but argument is of type ‘bfloat16 *’ {aka ‘short unsigned int *’}
make[1]: *** [Makefile:1936: cblas_sbgemmt.o] Error 1
````

--- a/interface/gemmt.c
+++ b/interface/gemmt.c
@@ -527,7 +527,7 @@ void CNAME(enum CBLAS_ORDER order, enum CBLAS_UPLO Uplo,
 #endif
 			// for alignment
 			buffer_size = (buffer_size + 3) & ~3;
-			STACK_ALLOC(buffer_size, FLOAT, buffer);
+			STACK_ALLOC(buffer_size, IFLOAT, buffer);
 
 #ifdef SMP
 
@@ -616,7 +616,7 @@ void CNAME(enum CBLAS_ORDER order, enum CBLAS_UPLO Uplo,
 #endif
 			// for alignment
 			buffer_size = (buffer_size + 3) & ~3;
-			STACK_ALLOC(buffer_size, FLOAT, buffer);
+			STACK_ALLOC(buffer_size, IFLOAT, buffer);
 
 #ifdef SMP
 

